app/controllers/CheckYourAnswersController.scala

```diff
@@ -169,81 +154,143 @@ class CheckYourAnswersController @Inject()(
+      case DataMissingError(EuSalesAtVatRateQuery(countryFromIndex, countryToIndex, vatRateIndex)) =>
+        logger.error(s"Data missing - net value of sales from country ${countryFromIndex.position} to country " +
+          s"${countryToIndex.position} at vat rate ${vatRateIndex.position} ")
+        Some(routes.NetValueOfSalesFromEuController.onPageLoad(CheckMode, period, countryFromIndex, countryToIndex, vatRateIndex))
+      case DataMissingError(VatOnSalesFromEuQuery(countryFromIndex, countryToIndex, vatRateIndex)) =>
+        logger.error(s"Data missing - vat charged on sales from country ${countryFromIndex.position} to country " +
+          s"${countryToIndex.position} at vat rate ${vatRateIndex.position} ")
+        Some(routes.VatOnSalesFromEuController.onPageLoad(CheckMode, period, countryFromIndex, countryToIndex, vatRateIndex))
+
+      case DataMissingError(AllCorrectionPeriodsQuery) =>
+        logger.error(s"Data missing - no data provided for corrections")
+        Some(correctionsRoutes.CorrectionReturnPeriodController.onPageLoad(CheckMode, period, Index(0)))
+      case DataMissingError(AllCorrectionCountriesQuery(periodIndex)) =>
+        logger.error(s"Data missing - no countries found for corrections to period ${periodIndex.position}")
+        Some(correctionsRoutes.CorrectionCountryController.onPageLoad(CheckMode, period, periodIndex, Index(0)))
+      case DataMissingError(CorrectionToCountryQuery(periodIndex, countryIndex)) =>
+        logger.error(s"Data missing - correction to country ${countryIndex.position} in period ${periodIndex.position}")
+        Some(correctionsRoutes.CountryVatCorrectionController.onPageLoad(CheckMode, period, periodIndex, countryIndex))
+
+      case DataMissingError(_) =>
+        logger.error(s"Unhandled DataMissingError")
+        None
+      case _ =>
+        logger.error(s"Unhandled ValidationError")
+        None
+    }
+  }
+
+  def submitReturn(vatReturnRequest: VatReturnRequest, correctionRequest: Option[CorrectionRequest], period: Period)
+                  (implicit request: DataRequest[AnyContent]): Future[Result] = {
+
+    val submission = correctionRequest match {
+      case Some(cr) => vatReturnConnector.submit(VatReturnWithCorrectionRequest(vatReturnRequest, cr))
+      case _ => vatReturnConnector.submit(vatReturnRequest)
+    }
+
+    submission.flatMap {
+      case Right(vatReturn: VatReturn) =>
+        auditEmailAndRedirect(vatReturnRequest, correctionRequest, vatReturn, period)
+      case Right((vatReturn: VatReturn, _)) =>
+        auditEmailAndRedirect(vatReturnRequest, correctionRequest, vatReturn, period)
+      case Left(ConflictFound) =>
+        auditService.audit(ReturnsAuditModel.build(
+          vatReturnRequest, correctionRequest, SubmissionResult.Duplicate, None, None, request
+        ))
+        Redirect(routes.YourAccountController.onPageLoad()).toFuture
```

Comment:
why do we go to your account page?

Reply:
Not sure, this was kept as is from main. Can change it if you'd like?